common/lib/dependabot/pull_request_creator/message_builder.rb

```diff
@@ -80,15 +83,17 @@ class MessageBuilder
           dependency_group: T.nilable(Dependabot::DependencyGroup),
           pr_message_max_length: T.nilable(Integer),
           pr_message_encoding: T.nilable(Encoding),
-          ignore_conditions: T::Array[T::Hash[String, String]]
+          ignore_conditions: T::Array[T::Hash[String, String]],
+          notices: T.nilable(T::Array[T::Hash[T.any(String, Symbol), T.untyped]])
```

Comment:
Let's make Notice a class so it can be typed. The T::Hash pattern is prone to typos.